tsdb/wal/wal.go

```diff
@@ -774,6 +784,52 @@ func listSegments(dir string) (refs []segmentRef, err error) {
 	return refs, nil
 }
 
+// Get segments from the directory
+func getSegments(dir string) ([]segmentRef, error) {
+	files, err := fileutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	var refs []segmentRef
+	for _, fn := range files {
+		k, err := strconv.Atoi(fn)
+		if err != nil {
+			return nil, err
+		}
+		refs = append(refs, segmentRef{name: fn, index: k})
+	}
+	sort.Slice(refs, func(i, j int) bool {
+		return refs[i].index < refs[j].index
+	})
+	return refs, nil
+}
+
+// RepairUnsequentialSegments repairs the out of sequence segments by deleting the onces after the sequence gap.
+// Returns the sequential segments after deleting the out of sequence wals after the sequence gap.
+func (w *WAL) RepairUnsequentialSegments(dir string) error {
+	segments, err := getSegments(dir)
+	if err != nil {
+		return errors.Wrap(err, "getSegments")
+	}
+	last := -1
```

Comment:
Segments won't start with 0 always, this should be set to the 1st segment and start the loop from 2nd segment, else it will end up deleting all the segments according to the below logic,